src/services/subsocial/evmAddresses/query.ts

```diff
@@ -1,16 +1,35 @@
-import { SubsocialQueryData, createSubsocialQuery } from '@/subsocial-query/subsocial/query'
+import { AccountData } from '@/pages/api/accounts-data'
+import { createQuery, poolQuery } from '@/subsocial-query'
 
-const getLinkedEvmAddress = async (params: SubsocialQueryData<string>): Promise<string> => {
-  const { api: subsocialApi, data } = params
+import axios from 'axios'
 
-  const api = await subsocialApi.blockchain.api
+export async function getAccountsData(addresses: string[]) {
+  const requestedIds = addresses.filter((id) => !!id)
+  if (requestedIds.length === 0) return []
+  const res = await axios.get(
+    '/api/accounts-data?' + requestedIds.map((n) => `addresses=${n}`).join('&')
+  )
 
-  const linkedAddresses = await api.query.evmAccounts.evmAddressByAccount(data)
+  return res.data.data as AccountData[]
+}
+
+export async function mutateAccountsDataCache(address: string) {
```

Comment:
its better to move this to mutation, where its used, and not export it because its only used there